qiita_db/commands.py

```diff
@@ -300,3 +304,101 @@ def load_parameters_from_cmd(name, fp, table):
                          "The format is PARAMETER_NAME<tab>VALUE")
 
     return constructor.create(name, **params)
+
+
+def update_preprocessed_data_from_cmd(sl_out_dir, study_id):
+    """Updates the preprocessed data of the study 'study_id'
+
+    Parameters
+    ----------
+    sl_out_dir : str
+        The path to the split libraries output directory
+    study_id : int
+        The study_id of the study to be updated
+
+    Returns
+    -------
+    qiita_db.PreprocessedData
+        The updated preprocessed data
+
+    Raises
+    ------
+    ValueError
+        If sl_out_dir does not contain all the required files
+        If the study does not have any preprocessed data
+
+    Notes
+    -----
+    If the study has more than one preprocessed data, the one with lower id
+    is updated.
+    """
+    # Check that we have all the required files
+    path_builder = partial(join, sl_out_dir)
+    new_fps = {'preprocessed_fasta': path_builder('seqs.fna'),
+               'preprocessed_fastq': path_builder('seqs.fastq'),
+               'preprocessed_demux': path_builder('seqs.demux'),
+               'log': path_builder('split_library_log.txt')}
+
+    missing_files = [key for key, val in new_fps.items() if not exists(val)]
```

Comment:
both appear throughout the codebase, so I don't think it's blocking, but I do agree that we should be using `viewitems` wherever possible